js/search/search.js

```diff
@@ -66,94 +46,94 @@ class Search {
+                let indexDateStartFrom = new Date().getTime() - searchConfig.SEARCH_PERIOD_SUBTRACTOR;
+                // Deleting all the messages except 3 Months from now
+                libSymphonySearch.symSEDeleteMessagesFromRAMIndex(null,
+                    searchConfig.MINIMUM_DATE, indexDateStartFrom.toString());
+            });
+        }
+    }
+
+    deCompress(key) {
+        const userIndexPath = path.join(searchConfig.FOLDERS_CONSTANTS.INDEX_PATH,
+            `${searchConfig.FOLDERS_CONSTANTS.PREFIX_NAME}_${this.userId}`);
+        if (fs.existsSync(`${userIndexPath}${searchConfig.TAR_LZ4_EXT}`)) {
+            lz4.deCompression(`${userIndexPath}${searchConfig.TAR_LZ4_EXT}`, (err) => {
+                if (err && !fs.existsSync(userIndexPath)) {
+                    fs.mkdirSync(userIndexPath);
+                }
+                this.init(key);
+            })
+        } else {
+            if (!fs.existsSync(userIndexPath)) {
+                fs.mkdirSync(userIndexPath);
+            }
+            this.init(key);
+        }
     }
 
     /**
      * An array of messages is passed for indexing
      * it will be indexed in a temporary index folder
      * @param {Array} messages
-     * @returns {Promise}
+     * @param callback
      */
-    indexBatch(messages) {
-        return new Promise((resolve, reject) => {
-            if (!messages) {
-                log.send(logLevels.ERROR, 'Batch Indexing: Messages not provided');
-                reject(new Error('Batch Indexing: Messages are required'));
-                return;
-            }
-
-            try {
-                let msg = JSON.parse(messages);
-                if (!(msg instanceof Array)) {
-                    log.send(logLevels.ERROR, 'Batch Indexing: Messages must be an array');
-                    reject(new Error('Batch Indexing: Messages must be an array'));
-                    return;
-                }
-            } catch(e) {
-                log.send(logLevels.ERROR, 'Batch Indexing: parse error -> ' + e);
-                reject(new Error(e));
-                return;
-            }
+    indexBatch(messages, callback) {
+        if (typeof callback !== "function") {
+            return false;
+        }
 
-            if (!this.isInitialized) {
-                log.send(logLevels.ERROR, 'Library not initialized');
-                reject(new Error('Library not initialized'));
-                return;
-            }
+        if (!messages) {
+            log.send(logLevels.ERROR, 'Batch Indexing: Messages not provided');
+            return callback(false, 'Batch Indexing: Messages are required');
+        }
 
-            if (!fs.existsSync(searchConfig.FOLDERS_CONSTANTS.INDEX_PATH)) {
-                log.send(logLevels.ERROR, 'User index folder not found');
-                reject(new Error('User index folder not found'));
-                return;
+        try {
+            let msg = JSON.parse(messages);
```

Comment:
You can probably rename this to something like `messagesData`. `msg` gives an impression that it's an individual message and not a set of them.